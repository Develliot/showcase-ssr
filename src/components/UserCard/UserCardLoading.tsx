import React from 'react';

import { VerticalSpacer, RowCenter } from 'src/components/Layout';

import {
    UserCardWrapper,
    BottomHalf,
    LoadingCoverColor,
    LoadingImageWrapper,
    LoadingH3,
    LoadingParagraph,
    AnimatedRows,
} from './styles';

export const UserCardLoading = () => {
    return (
        <UserCardWrapper>
            <RowCenter>
                <LoadingCoverColor>
                    <RowCenter>
                        <LoadingImageWrapper />
                    </RowCenter>
                </LoadingCoverColor>
            </RowCenter>
            <BottomHalf>
                <VerticalSpacer size='small' />
                <AnimatedRows>
                    <LoadingH3 />
                </AnimatedRows>
                <VerticalSpacer size='medium' />
                <AnimatedRows>
                    <LoadingParagraph />
                    <LoadingParagraph />
                    <LoadingParagraph />
                    <LoadingParagraph />
                    <LoadingParagraph />
                </AnimatedRows>
            </BottomHalf>
        </UserCardWrapper>
    );
};
