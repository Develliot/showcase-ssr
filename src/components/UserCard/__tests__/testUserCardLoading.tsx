import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { UserCardLoading } from '../UserCardLoading';

it('renders correctly', () => {
    const wrapper: ShallowWrapper = shallow(<UserCardLoading />);
    expect(wrapper).toMatchSnapshot();
});
