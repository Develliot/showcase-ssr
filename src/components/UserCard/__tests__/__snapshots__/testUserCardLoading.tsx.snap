// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styles__UserCardWrapper>
  <Layout__RowCenter>
    <styles__LoadingCoverColor>
      <Layout__RowCenter>
        <styles__LoadingImageWrapper />
      </Layout__RowCenter>
    </styles__LoadingCoverColor>
  </Layout__RowCenter>
  <styles__BottomHalf>
    <Layout__VerticalSpacer
      size="small"
    />
    <styles__AnimatedRows>
      <styles__LoadingH3 />
    </styles__AnimatedRows>
    <Layout__VerticalSpacer
      size="medium"
    />
    <styles__AnimatedRows>
      <styles__LoadingParagraph />
      <styles__LoadingParagraph />
      <styles__LoadingParagraph />
      <styles__LoadingParagraph />
      <styles__LoadingParagraph />
    </styles__AnimatedRows>
  </styles__BottomHalf>
</styles__UserCardWrapper>
`;
